src/sidebar/store/debug-middleware.js

```diff
@@ -7,14 +12,17 @@
  * When enabled, every action that changes application state will be printed
  * to the console, along with the application state before and after the action
  * was handled.
+ *
+ * @param {import("redux").Store} store
  */
 export default function debugMiddleware(store) {
   /* eslint-disable no-console */
   let serial = 0;
 
   return function (next) {
     return function (action) {
-      if (!window.debug) {
+      // @ts-ignore The window interface needs to be expanded to include this property
```

Comment:
oops, forgot to remove!

Reply:
intersections just don;'t seem to work, might be related to this ticket
https://github.com/jsdoc/jsdoc/issues/1285


Leaving the @ignore for now and I'll remove the casting

Reply:
Ah, I see the confusion. That ticket is on the _JSDoc_ repository, rather than the [TypeScript one](https://github.com/Microsoft/TypeScript/). The original JSDoc spec was written for human readers, so if you're looking at the [JSDoc website](https://jsdoc.app) that doesn't include all the types that TS understands.

Here is an example of where I have successfully used intersection types: https://github.com/hypothesis/client/blob/478772e34fdebabc67da556cc53efe290461c74a/src/annotator/config/config-func-settings-from.js#L21

In any case, it is fine to ignore this for now and come back to it later.

Reply:
okay, thanks for the clarification